test/e2e-layering/layering_test.go

```diff
@@ -1,7 +1,171 @@
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/daemon/constants"
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/openshift/machine-config-operator/test/helpers"
+)
+
+type LayeringSuite struct {
+	suite.Suite
+}
+
+func (suite *LayeringSuite) SetupSuite() {
+	cs := framework.NewClientSet("")
+
+	// don't keep unlabel function or delete pool function since removing a node from a layered pool is...not well defined
+	helpers.LabelAllNodesInPool(suite.T(), cs, "worker", "node-role.kubernetes.io/"+ctrlcommon.ExperimentalLayeringPoolName)
+	helpers.CreateMCP(suite.T(), cs, ctrlcommon.ExperimentalLayeringPoolName)
+
+	// TODO wait for that to rollout and assert that it rebased
+}
+
+func (suite *LayeringSuite) TestNoReboot() {
+	cs := framework.NewClientSet("")
+	unlabelFunc := helpers.LabelRandomNodeFromPool(suite.T(), cs, "worker", "node-role.kubernetes.io/infra")
+	oldInfraRenderedConfig := helpers.GetMcName(suite.T(), cs, "infra")
+
+	infraNode := helpers.GetSingleNodeByRole(suite.T(), cs, "infra")
+
+	output := helpers.ExecCmdOnNode(suite.T(), cs, infraNode, "cat", "/rootfs/proc/uptime")
+	oldTime := strings.Split(output, " ")[0]
+	suite.T().Logf("Node %s initial uptime: %s", infraNode.Name, oldTime)
+
+	// Adding authorized key for user core
+	testIgnConfig := ctrlcommon.NewIgnConfig()
+	testSSHKey := ign3types.PasswdUser{Name: "core", SSHAuthorizedKeys: []ign3types.SSHAuthorizedKey{"test adding authorized key without node reboot"}}
+	testIgnConfig.Passwd.Users = append(testIgnConfig.Passwd.Users, testSSHKey)
+
+	addAuthorizedKey := &mcfgv1.MachineConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   fmt.Sprintf("authorzied-key-infra-%s", uuid.NewUUID()),
+			Labels: helpers.MCLabelForRole("infra"),
+		},
+		Spec: mcfgv1.MachineConfigSpec{
+			Config: runtime.RawExtension{
+				Raw: helpers.MarshalOrDie(testIgnConfig),
+			},
+		},
+	}
+
+	_, err := cs.MachineConfigs().Create(context.TODO(), addAuthorizedKey, metav1.CreateOptions{})
+	require.Nil(suite.T(), err, "failed to create MC")
+	suite.T().Logf("Created %s", addAuthorizedKey.Name)
+
+	// grab the latest worker- MC
+	renderedConfig, err := helpers.WaitForRenderedConfig(suite.T(), cs, "infra", addAuthorizedKey.Name)
+	require.Nil(suite.T(), err)
+	err = helpers.WaitForPoolComplete(suite.T(), cs, "infra", renderedConfig)
+	require.Nil(suite.T(), err)
+
+	// Re-fetch the infra node for updated annotations
+	infraNode = helpers.GetSingleNodeByRole(suite.T(), cs, "infra")
+
+	assert.Equal(suite.T(), infraNode.Annotations[constants.CurrentMachineConfigAnnotationKey], renderedConfig)
```

Comment:
@jkyros I think I broke this in `experimentalUpdateLayeredConfig`. What would an equivalent condition be? Not sure if newest is correct or not, but something like `Annotations[ExperimentalNewestLayeredImageEquivalentConfigAnnotationKey == renderedConfig && Annotations[ExperimentalNewestLayeredImageAnnotationKey] == Annotations[CurrentImageConfigAnnotationKey]`